sphinx/application.py

```diff
@@ -187,8 +185,8 @@ def __init__(self, srcdir, confdir, outdir, doctreedir, buildername,
 
         # read config
         self.tags = Tags(tags)
-        self.config = Config(self.confdir, CONFIG_FILENAME,
-                             confoverrides or {}, self.tags)
+        self.config = Config.read(self.confdir, CONFIG_FILENAME,
```

Comment:
I think this needs to be changed too. Previous this wouldn't doing anything if you didn't provide `confdir`. You've changed this now, so you want something like this:

    if self.confdir:
        self.config = Config.read(...)
    else:
        self.config = Config({}, confoverrides or {})

Correct me if I'm wrong.